relayer/relays/beacon/header/header.go

```diff
@@ -0,0 +1,262 @@
+package header
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	log "github.com/sirupsen/logrus"
+	"github.com/snowfork/go-substrate-rpc-client/v4/types"
+	"github.com/snowfork/snowbridge/relayer/relays/beacon/cache"
+	"github.com/snowfork/snowbridge/relayer/relays/beacon/header/syncer"
+	"github.com/snowfork/snowbridge/relayer/relays/beacon/header/syncer/scale"
+	"github.com/snowfork/snowbridge/relayer/relays/beacon/writer"
+)
+
+var ErrFinalizedHeaderUnchanged = errors.New("finalized header unchanged")
+
+type Header struct {
+	cache  *cache.BeaconCache
+	writer *writer.ParachainWriter
+	syncer *syncer.Syncer
+}
+
+func New(writer *writer.ParachainWriter, beaconEndpoint string, slotsInEpoch uint64, epochsPerSyncCommitteePeriod uint64) Header {
+	return Header{
+		cache:  cache.New(),
+		writer: writer,
+		syncer: syncer.New(beaconEndpoint, slotsInEpoch, epochsPerSyncCommitteePeriod),
+	}
+}
+
+func (h *Header) Sync(ctx context.Context) (<-chan uint64, <-chan uint64, error) {
```

Comment:
Should also pass in the associated errgroup to spawn a goroutine (see my comment below).